File: nationsFrontEnd/src/app/shared/models/country.ts
export class Country{

  constructor(private _countryId?: number,
              private _name?: string,
              private _area?: number,
              private _nationalDay?: Date,
              private _countryCode2?: string,
              private _countryCode3?: string,
              private _regionId?: number) {
  }


  get countryId(): number {
    return this._countryId;
  }

  set countryId(value: number) {
    this._countryId = value;
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get area(): number {
    return this._area;
  }

  set area(value: number) {
    this._area = value;
  }

  get nationalDay(): Date {
    return this._nationalDay;
  }

  set nationalDay(value: Date) {
    this._nationalDay = value;
  }

  get countryCode2(): string {
    return this._countryCode2;
  }

  set countryCode2(value: string) {
    this._countryCode2 = value;
  }

  get countryCode3(): string {
    return this._countryCode3;
  }

  set countryCode3(value: string) {
    this._countryCode3 = value;
  }

  get regionId(): number {
    return this._regionId;
  }

  set regionId(value: number) {
    this._regionId = value;
  }
}

File: nationsBackEnd/src/main/java/com/qualco/nations/models/Country.java
package com.qualco.nations.models;

import lombok.Data;

import javax.persistence.*;
import java.io.Serializable;
import java.math.BigDecimal;
import java.time.LocalDate;

@Data
@Entity
@Table(name = "countries")
public class Country implements Serializable {

    private static final long serialVersionUID = 1L;

    @Id
    @Column(name = "country_id", nullable = false)
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer countryId;

    @Column(name = "name")
    private String name;

    @Column(name = "area", nullable = false)
    private BigDecimal area;

    @Column(name = "national_day")
    private LocalDate nationalDay;

    @Column(name = "country_code2", nullable = false)
    private String countryCode2;

    @Column(name = "country_code3", nullable = false)
    private String countryCode3;

    @Column(name = "region_id", nullable = false)
    private Integer regionId;

}
